packages/assistant-stream/src/core/utils/stream/merge.ts

```
@@ -0,0 +1,215 @@
+  reader: ReadableStreamDefaultReader<AssistantStreamChunk>;
+  promise?: Promise<unknown> | undefined;
+};
+
+const promiseWithResolvers = () => {
+  let resolve: () => void;
+  let reject: (reason?: any) => void;
+  const promise = new Promise<void>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return { promise, resolve: resolve!, reject: reject! };
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)